src/usr/local/www/status_queues.php

```diff
@@ -136,13 +85,125 @@ function getqueueactivity() {
+					interfacename_stats[interfacename][statqname]['suspends'] = suspends;
+					interfacename_stats[interfacename][statqname]['droppedpkts'] = droppedpkts;
+					interfacename_stats[interfacename][statqname]['qlengthitems'] = queue['qlengthitems'];
+					interfacename_stats[interfacename][statqname]['qlengthsize'] = queue['qlengthsize'];
+					find = queuename;
+					while(queueparents[find]) {
+						// add diff values also to parent queues
+						parentname = queueparents[find];
+						parentqueuename = parentname+interfacename;
+						interfacename_stats[interfacename][parentqueuename]['pkts_ps'] += pkts_ps;
+						interfacename_stats[interfacename][parentqueuename]['bytes_ps'] += bytes_ps;
+						interfacename_stats[interfacename][parentqueuename]['borrows'] += borrows;
+						interfacename_stats[interfacename][parentqueuename]['suspends'] += suspends;
+						interfacename_stats[interfacename][parentqueuename]['droppedpkts'] += droppedpkts;
+						find = parentname;
+					}
+				}
+				queuestatprevious[statqname] = queue; 
+			}
+		}
+		// Find max pps/bps needed for any scale bar
+		statmax = 0;
+		for (interfacename in interfacename_stats) {
+			interface = interfacename_stats[interfacename];
+			for (queuename in interface) {
+				queue = interface[queuename];
+				if ($stattype == "0") {
+					if (statmax < queue['pkts_ps']) {
+						statmax = queue['pkts_ps'];
+					}
+				} else {
+					if (statmax < queue['bytes_ps']) {
+						statmax = queue['bytes_ps'];
+					}
+				}
+			}
+		}
+		// use a slowly sliding max scale value but do make sure its always large enough to accomodate the largest value..
+		if (graphstatmax < statmax) {
+			// peek value + 10% keeps a little room for it to increase
+			graphstatmax = statmax * 1.1;
+		} else {
+			// in general make largest bar fill +- 2/3 of the scale
+			graphstatmax = (graphstatmax * 20 + statmax * 1.5) / 21;
+		}
+		// set values on the objects
+		for (interfacename in interfacename_stats) {
+			interface = interfacename_stats[interfacename];
+			for (queuename in interface) {
+				queue = interface[queuename];
+				statqname = queuename;
+				if ($stattype == "0") {
+					$('#queue'+statqname+'width').css('width', (queue['pkts_ps']*100/graphstatmax).toFixed(0) + '%');
+				} else {
+					$('#queue'+statqname+'width').css('width', (queue['bytes_ps']*100/graphstatmax).toFixed(0) + '%');
+				}
+				$('#queue'+statqname+'pps').val(queue['pkts_ps'].toFixed(1));
+				$('#queue'+statqname+'bps').val(formatSpeedBits(queue['bytes_ps']));
+				$('#queue'+statqname+'borrows').val(queue['borrows']);
+				$('#queue'+statqname+'suspends').val(queue['suspends']);
+				$('#queue'+statqname+'drops').val(queue['droppedpkts']);
+				$('#queue'+statqname+'length').val(queue['qlengthitems']+'/'+queue['qlengthsize']);
+			}
+		}
+		timestampprevious = timestamp;   
 	}
 
 	$(document).ready(function() {
 		setTimeout(getqueueactivity, 150);
 	});
 });
+
+function formatSpeedBits(speed) {
```

Comment:
It seems difficult to move.. <?=gettext("Kbps"); ?> doesnt work in pfSenseHelper.js as thats not parsed through php.. so maybe this is the best spot after all. even though it does duplicate javascript code thats also in graph.php. not sure what would be the proper way do de-duplicate this code..